mapper/orm/src/main/java/org/hibernate/search/mapper/orm/automaticindexing/impl/AutomaticIndexingQueueEventProcessingPlanImpl.java

```diff
@@ -45,4 +46,17 @@ public void delete(String entityName, String serializedId, DocumentRoutesDescrip
 	public CompletableFuture<MultiEntityOperationExecutionReport<EntityReference>> executeAndReport() {
 		return delegate.executeAndReport( entityReferenceFactory );
 	}
+
+	@Override
+	public String toSerializedId(String entityName, Object identifier) {
+		return delegate.toSerializedId( entityName, identifier );
+	}
+
+	@Override
+	public EntityReference entityReference(String entityName, String serializedId) {
+		PojoIndexingQueueEventProcessingPlan.EntityReferenceInfo refInfo =
+				delegate.entityReference( entityName, serializedId );
```

Comment:
Instead of using an intermediary representation (`EntityReferenceInfo`), you could pass the `entityReferenceFactory` to `delegate.entityReference()` so that it returns the right type of reference right away.

That's how it's done everywhere else, at least :)